services/agents/postgresql.go

```diff
@@ -0,0 +1,76 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package agents
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"sort"
+	"strconv"
+
+	"github.com/AlekSi/pointer"
+	"github.com/percona/pmm/api/agentpb"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+func postgresqlDSN(service *models.Service, exporter *models.Agent) string {
+	q := make(url.Values)
+	q.Set("sslmode", "disable") // TODO https://jira.percona.com/browse/PMM-1727
+	q.Set("connect_timeout", strconv.Itoa(5))
+
+	address := net.JoinHostPort(*service.Address, strconv.Itoa(int(*service.Port)))
+	uri := url.URL{
+		Scheme:   "postgres",
+		User:     url.UserPassword(*exporter.Username, *exporter.Password),
+		Host:     address,
+		Path:     "postgres",
+		RawQuery: q.Encode(),
+	}
+	return uri.String()
+}
+
+// postgresExporterConfig returns desired configuration of mysqld_exporter process.
+func postgresExporterConfig(service *models.Service, exporter *models.Agent) *agentpb.SetStateRequest_AgentProcess {
+	tdp := templateDelimsPair(
+		pointer.GetString(service.Address),
+		pointer.GetString(exporter.Username),
+		pointer.GetString(exporter.Password),
+		pointer.GetString(exporter.MetricsURL),
+	)
+
+	args := []string{
+		"-web.listen-address=:" + tdp.left + " .listen_port " + tdp.right,
```

Comment:
I _think_ postgres_exporter uses`--` for all flags. Please check it.

Reply:
Checked, it uses only one `-` for flags